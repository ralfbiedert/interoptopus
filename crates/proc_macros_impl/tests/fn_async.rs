use quote::quote;
use syn::{parse_quote, ItemFn};

mod util;

// TODO this should fail somehow and we want to check that
#[test]
fn async_fails() {
    let item: ItemFn = parse_quote! {
        #[ffi_function]
        async fn async_foo() {}
    };

    test_fn!(item);
}
