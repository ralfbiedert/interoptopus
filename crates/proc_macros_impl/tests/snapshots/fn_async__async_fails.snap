---
source: crates/proc_macros_impl/tests/fn_async.rs
expression: formatted
---
#[unsafe(export_name = "async_foo")]
extern "C" fn async_foo() {}
#[allow(non_camel_case_types)]
struct async_foo {}
impl ::interoptopus::lang::function::FunctionInfo for async_foo {
    fn id() -> ::interoptopus::inventory::FunctionId {
        ::interoptopus::inventory::FunctionId::from_id(::interoptopus::id!(async_foo))
    }
    fn signature() -> ::interoptopus::lang::function::Signature {
        ::interoptopus::lang::function::Signature {
            arguments: vec![],
            rval: <() as ::interoptopus::lang::types::TypeInfo>::id(),
        }
    }
    fn function() -> ::interoptopus::lang::function::Function {
        ::interoptopus::lang::function::Function {
            name: "async_foo".to_string(),
            visibility: ::interoptopus::lang::meta::Visibility::Private,
            docs: ::interoptopus::lang::meta::Docs::from_lines(vec![]),
            emission: ::interoptopus::lang::meta::Emission::External,
            signature: Self::signature(),
        }
    }
    fn register(inventory: &mut ::interoptopus::inventory::Inventory) {
        <() as ::interoptopus::lang::types::TypeInfo>::register(inventory);
        inventory.register_function(Self::id(), Self::function());
    }
}
